compiler/noirc_evaluator/src/ssa/parser.rs

```diff
@@ -0,0 +1,792 @@
+        };
+
+        self.bump()?;
+
+        Ok(Some(op))
+    }
+
+    fn parse_call(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::Call)? {
+            return Ok(None);
+        }
+
+        let function = self.eat_identifier_or_error()?;
+        let arguments = self.parse_arguments()?;
+        Ok(Some(ParsedInstruction::Call { targets: vec![], function, arguments, types: vec![] }))
+    }
+
+    fn parse_constrain(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::Constrain)? {
+            return Ok(None);
+        }
+
+        let lhs = self.parse_value_or_error()?;
+        self.eat_or_error(Token::Equal)?;
+        let rhs = self.parse_value_or_error()?;
+        Ok(Some(ParsedInstruction::Constrain { lhs, rhs }))
+    }
+
+    fn parse_decrement_rc(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::DecRc)? {
+            return Ok(None);
+        }
+
+        let value = self.parse_value_or_error()?;
+        Ok(Some(ParsedInstruction::DecrementRc { value }))
+    }
+
+    fn parse_enable_side_effects(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::EnableSideEffects)? {
+            return Ok(None);
+        }
+
+        let condition = self.parse_value_or_error()?;
+        Ok(Some(ParsedInstruction::EnableSideEffectsIf { condition }))
+    }
+
+    fn parse_increment_rc(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::IncRc)? {
+            return Ok(None);
+        }
+
+        let value = self.parse_value_or_error()?;
+        Ok(Some(ParsedInstruction::IncrementRc { value }))
+    }
+
+    fn parse_range_check(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::RangeCheck)? {
+            return Ok(None);
+        }
+
+        let value = self.parse_value_or_error()?;
+        self.eat_or_error(Token::Keyword(Keyword::To))?;
+        let max_bit_size = self.eat_int_or_error()?.to_u128() as u32;
+        self.eat_or_error(Token::Keyword(Keyword::Bits))?;
+        Ok(Some(ParsedInstruction::RangeCheck { value, max_bit_size }))
+    }
+
+    fn parse_store(&mut self) -> ParseResult<Option<ParsedInstruction>> {
+        if !self.eat_keyword(Keyword::Store)? {
+            return Ok(None);
+        }
+
+        let value = self.parse_value_or_error()?;
+        self.eat_or_error(Token::Keyword(Keyword::At))?;
+        let address = self.parse_value_or_error()?;
+        Ok(Some(ParsedInstruction::Store { address, value }))
+    }
+
+    fn parse_assignment(&mut self, target: Identifier) -> ParseResult<Option<ParsedInstruction>> {
+        let mut targets = vec![target];
+
+        while self.eat(Token::Comma)? {
+            let target = self.eat_identifier_or_error()?;
+            targets.push(target);
+        }
+
+        self.eat_or_error(Token::Assign)?;
```

Comment:
So at this point we're sure that we are dealing with `<target> = `, right? After that an `Ok(None)` value wouldn't be valid, would it?